File: openshmem-book/programs/hello-openshmem.c
#include <stdio.h>
#include <shmem.h>          /* The shmem header file */

int
main (int argc, char *argv[])
{
  int nprocs, me;

  start_pes (0);
  nprocs = shmem_n_pes ();
  me = shmem_my_pe ();
  printf ("Hello from %d of %d\n", me, nprocs);
  return 0;
}

File: openshmem-book/programs/hello-openshmem.f90
program hello

  include 'shmem.fh'
  integer :: shmem_my_pe, shmem_n_pes

  integer :: npes, me

  call start_pes (0)
  npes = shmem_n_pes ()
  me = shmem_my_pe ()

  write (*, 1000) me, npes

 1000 format ('Hello from', 1X, I4, 1X, 'of', 1X, I4)

end program hello
